docs/integration-guides/key-management.md

```diff
@@ -81,7 +81,14 @@ If the need arises to manually trigger a block confirmation, either due to missi
 
 External accounting systems that track balances arriving to the node must track hashes of blocks that have been received in order to guarantee idempotency. Once confirmation of a block has been validated, the block hash should be recorded for the account along with any credits, debits or other related information. Any attempts to credit or debit accounts external to the node should check that no previous conflicting or duplicate activity was already recorded for that same block hash.
 
+#### Transaction order and correctness
+
+Transactions do not directly specify the amount of funds that will be moved, only the remaining balance after the transaction. When creating and processing blocks, ensure that you are the only entity with access to the account's private key, and no two processes are attempting to create blocks for the same account simultaneously. Failing to ensure these conditions could result in the wrong amount being transacted.
```

Comment:
After reviewing this more, I wonder if we should include this: `When creating and processing blocks, ensure that you are the only entity with access to the account's private key, and no two processes are attempting to create blocks for the same account simultaneously. Failing to ensure these conditions could result in the wrong amount being transacted.` The general push for an atomic call on `account_info` means that even if someone else was doing stuff on the account, then a transactions created with only details from that atomic call could not accidentally spend the wrong amount - it could point to the wrong previous block, but then would be invalid.

Reply:
Ok, think we found a good balance. I've also modified the note on RPCs `account[s]_balance[s]` to say "prefer `account_info` for creating transactions".